lib/WebService/Async/CustomerIO.pm

```diff
@@ -147,15 +139,47 @@ Getter returns RateLimmiter for regular API endpoint.
 
 =cut
 
-sub api_ratelimiter {shift->{api_ratelimiter}}
+sub api_ratelimiter {
+    my ($self) = @_;
+
+    return $self->{api_ratelimiter} if $self->{api_ratelimiter};
+
+    Carp::croak "Can't use rate limiter without a loop" unless $self->loop;
+
+    $self->{api_ratelimiter} =
+        WebService::Async::CustomerIO::RateLimiter->new(
+            limit    => REQUEST_PER_SECOND_LIMIT_API,
+            interval => 1,
+        );
+
+    $self->add_child($self->{api_ratelimiter});
```

Comment:
If I understand Tom's [comment](https://github.com/binary-com/perl-WebService-Async-CustomerIO/pull/1/files#r337333233) right, we should initialize this in a `sub _add_to_loop {...}`, for example like here: https://github.com/regentmarkets/bom-events/blob/master/lib/BOM/Event/QueueHandler.pm#L112-L116

Reply:
We discussed with @tom-binary  this point offline. The getter is also appropriate place we just need to check that we have loop, when we're trying to access to rate limiter. The main point, because rate limitter depends on loop in few places, and to prevent falling inside code of rate limiter, we need to be sure that we have loop before using it.

Reply:
OK thanks for the explanation